src/scala/com/github/johnynek/bazel_deps/DepsModel.scala

```diff
@@ -381,8 +394,15 @@ object Language {
   implicit val ordering: Ordering[Language] = Ordering.by(_.asString)
 }
 
-case class UnversionedCoordinate(group: MavenGroup, artifact: MavenArtifactId) {
-  def asString: String = s"${group.asString}:${artifact.asString}"
+case class UnversionedCoordinate(group: MavenGroup, artifact: MavenArtifactId, classifier: Option[Classifier]) {
+
+  override lazy val hashCode: Int =
```

Comment:
you don't need to do this. scala case classes have automatic hashCode implementations wihcih work basically like what you have (it also includes the name of the class).

Reply:
Will remove, thanks. I think this was leftover from trying to address a what I thought was a perf issue, but that turned out to be the infinite loop in Normalizer.